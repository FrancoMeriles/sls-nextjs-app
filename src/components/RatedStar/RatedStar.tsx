import { Box } from '@chakra-ui/react'
import { getStarsToRate } from '@base/utils'
import { Star } from '@base/icons'

const RatedStar = ({ rateAverage, boxSize }) => {
  const rates = getStarsToRate(rateAverage)
  return (
    <Box>
      {rateAverage > 0 &&
        rates.map((rateToStar, index) => (
          <Star boxSize={boxSize} color="brand.300" key={index} paint={rateToStar} />
        ))}
    </Box>
  )
}

export default RatedStar
