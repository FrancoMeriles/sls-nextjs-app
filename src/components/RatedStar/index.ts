import RatedStar from './RatedStar'

export default RatedStar
