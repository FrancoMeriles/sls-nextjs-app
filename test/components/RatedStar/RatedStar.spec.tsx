import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import RatedStar from '@base/components/RatedStar'

let ratingAverage = 3.8
const boxSize = 2

describe('RatedStar', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer
      .create(<RatedStar rateAverage={ratingAverage} boxSize={boxSize} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    const { container } = render(<RatedStar rateAverage={ratingAverage} boxSize={boxSize} />)
    expect(container.getElementsByClassName('chakra-icon').length).toBe(5)
  })

  it('should not renders if there is not ratingAverage', async () => {
    ratingAverage = 0
    const { container } = render(<RatedStar rateAverage={ratingAverage} boxSize={boxSize} />)
    expect(container.getElementsByClassName('chakra-icon').length).toBe(0)
  })
})
