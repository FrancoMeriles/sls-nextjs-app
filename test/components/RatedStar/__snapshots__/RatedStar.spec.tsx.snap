// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatedStar should render and match snapshot with required props 1`] = `
<div
  className="css-0"
>
  <svg
    className="chakra-icon css-q1ives"
    focusable={false}
    paint="full"
    viewBox="0 0 10 10"
  >
    <path
      d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
      fill="currentColor"
    />
  </svg>
  <svg
    className="chakra-icon css-q1ives"
    focusable={false}
    paint="full"
    viewBox="0 0 10 10"
  >
    <path
      d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
      fill="currentColor"
    />
  </svg>
  <svg
    className="chakra-icon css-q1ives"
    focusable={false}
    paint="full"
    viewBox="0 0 10 10"
  >
    <path
      d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
      fill="currentColor"
    />
  </svg>
  <svg
    className="chakra-icon css-q1ives"
    focusable={false}
    paint="half"
    viewBox="0 0 10 10"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        d="M5.256 8L2.131 9.648l.597-3.49L.2 3.684l3.494-.509L5.256 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
        fill="#e0e0e0"
      />
      <path
        d="M5.272 8.026L2.137 9.679l.6-3.502L.2 3.697l3.505-.51L5.272 0z"
        fill="currentColor"
      />
    </g>
  </svg>
  <svg
    className="chakra-icon css-q1ives"
    focusable={false}
    paint="empty"
    viewBox="0 0 10 10"
  >
    <path
      d="M5.056 8L1.931 9.648l.597-3.49L0 3.684l3.494-.509L5.056 0l1.562 3.176 3.494.51-2.528 2.471.597 3.491z"
      fill="#e0e0e0"
    />
  </svg>
</div>
`;
